plugins/dashboard/dashboard.js

```diff
@@ -0,0 +1,348 @@
+  this.config = options;
+
+  this.Q = require('q');
+
+  /**
+   * Basic authentication helper function.
+   */
+  this.auth = function auth(req, res, next) {
+    var fs = require('fs');
+    var basicAuth = require('basic-auth');
+    var htpasswd = require('htpasswd-auth');
+
+    function unauthorized(res) {
+      res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+      return res.sendStatus(401);
+    }
+
+    var user = basicAuth(req);
+
+    if (!user || !user.name || !user.pass) {
+      return unauthorized(res);
+    }
+
+    var file = fs.readFileSync(__dirname + '/../../' + self.config.htpasswd, 'utf8');
+    htpasswd.authenticate(user.name, user.pass, file).then(function (auth) {
+      if (auth) {
+        return next();
+      }
+      else {
+        return unauthorized(res);
+      }
+    });
+  };
+
+  // Interpreted and loaded here as it will be used many times.
+  this.bashboardTemplate = twig.twig({
+    data: fs.readFileSync(__dirname + '/views/status.html', 'utf8')
+  });
+
+  /**
+   * Dashboard status callback.
+   */
+  app.get('/dashboard/status', self.auth, function (req, res) {
+    self.buildScreenData().then(function (screens) {
+      res.send(self.bashboardTemplate.render({
+        page_title: 'Aroskanalen screen',
+        date: {
+          month: self.moment().format('MMMM'),
+          day: self.moment().format('D'),
+          year: self.moment().format('YYYY')
+        },
+        screens: screens
+      }));
+    },
+    function error(error) {
+      res.status(500).send(error.message);
+    })
+  });
+
+  /**
+   * Configuration page to black list screens form the critical list.
+   *
+   * This could be used to hide develeopment/test screens from the overview.
+   */
+  app.get('/dashboard/blacklist', self.auth, function (req, res) {
+    // Create local every time to not use global memory for the template.
+    var template = twig.twig({
+      data: fs.readFileSync(__dirname + '/views/blacklist.html', 'utf8')
+    });
+
+    self.buildScreenData().then(function (screens) {
+      self.load().then(function (blacklist) {
+        res.send(template.render({
+          screens: screens,
+          blacklist: blacklist
+        }));
+      });
+    },
+    function error(error) {
+      res.status(500).send(error.message);
+    })
+  });
+
+  /**
+   * Ajax callback function to update the back list json file.
+   */
+  app.post('/dashboard/blacklist/save', self.auth, function (req, res) {
+    var sortedList = {};
+    var list = req.body;
+    for (var i in list) {
+      var screen = list[i];
+      if (!sortedList.hasOwnProperty(screen.apikey)) {
+        sortedList[screen.apikey] = [];
+      }
+      sortedList[screen.apikey].push(screen.id);
+    }
+    self.save(sortedList);
+    res.json({});
+  });
+};
+
+/**
+ * Helper function to async load all information about screens known to the middleware.
+ */
+Dashboard.prototype.buildScreenData = function buildScreenData() {
```

Comment:
No, I don't want to touch it.

It's not that complex it's just a promise chain of dependencies. If we had use bluebird and not Q this might have been easier to read.

Renamed variables to more correct names and add more documentation :-)